### PR TITLE
feat: add CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by EstebanBAR0N**

### CODE REVIEW

**Review by iverly — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
+# Contributing to FAAST
+
+Thanks for your interest in improving the project! This document provides a step-by-step guide for general contributions to FAAST.
```
Comment (lucido-simon) [inline CONTRIBUTING.md:3]:
```suggestion
Thanks for your interest in improving the project! This document provides a guideline for general contributions to FAAST.
```
This is not a step by step guide. We just give the rules to follow.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+There are there general ways you can contribute to this repo:
+
+- Proposing an enhancement or new feature
+- Reporting a bug or regression
+- Contributing changes to the source code
+
+For the first two, refer to the [GitHub Issues](https://github.com/faast-rt/lambdo/issues/new/choose) which guides you through the available options along with the needed information to collect.
+
+> To setup your development environment, please refer to the [README](./README.md).
```
Comment (lucido-simon) [inline CONTRIBUTING.md:15]: Not sure why this is here. Would make more sense in "Contributing changes"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+## Contributing changes
+
+_Prior to opening a pull request, it is recommended to open an issue first to ensure the maintainers can review intended changes. Exceptions to this rule include fixing non-functional source such as code comments, documentation or other supporting files._
```
[the remark this answers is not in the document]
Comment (lucido-simon) [reply CONTRIBUTING.md:19]: We might be able to slightly change the wording to make the message clearer:

```suggestion
_Please open an issue before adding a new feature or making major changes to existing one. It allows the community to discuss the relevancy of the issue, as well as the caveats one should avoid when implementing it._
```
(Far from perfect, but gets the point across more accurately)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+Proposing source code changes is done through [GitHub's standard pull request workflow](https://docs.github.com/en/get-started/quickstart/github-flow#create-a-pull-request).
+
+If your branch is a work-in-progress then please start by creating your pull requests as draft, by clicking the down-arrow next to the `Create pull request` button and instead selecting `Create draft pull request`.
+
+This will defer the automatic process of requesting a review from the FAAST team and significantly reduces noise until you are ready. Once you are happy, you can click the `Ready for review` button.
+
```
Comment (lucido-simon) [inline CONTRIBUTING.md:26]: We don't talk where the contributor should contribute. I think they should fork the repo and create PRs from their repo. Adding a small sentence about that seems relevant.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+### Pull Request Guidelines
+
+A good pull request includes:
+
+- A high-level description of the changes, including links to any issues that are related by adding comments like `Resolves #NNN` to your description. See [Linking a Pull Request to an Issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for more information.
```
Comment (lucido-simon) [inline CONTRIBUTING.md:31]: Does it ? The high level description should be in the issue, not in the PR, right ? PR should discuss the implementation.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+To perform a sign-off with `git`, use :
+
+```bash
+git commit -s #(or --signoff)
+```
+
+## Get help
+
+If you have questions about the contribution process, please start a [GitHub discussion](https://github.com/faast-rt/lambdo/discussions) or send your question to the contact email address: [contact@faast-rt.com](mailto://contact@faast-rt.com).
```
Comment (lucido-simon) [inline CONTRIBUTING.md:135]: Do we really need all this text about conventional commits ? This is 2/3 of our contributing. We can have a small bullet point list to recap what it is about, but the reader is expected to read the specs if they don't know it.

File: CONTRIBUTING.md (hunk 2 above)
Comment (lucido-simon) [reply CONTRIBUTING.md:15]: (Comment applies to the last line of the suggestion)

File: CONTRIBUTING.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (WoodenMaiden) [reply CONTRIBUTING.md:19]: Sounds good

File: CONTRIBUTING.md (hunk 5 above)
Comment (alexis-langlet) [reply CONTRIBUTING.md:31]: Yes it does, you don't necessarily want to check all the issues to understand what a PR is about IMO

File: CONTRIBUTING.md (hunk 6 above)
Comment (iverly) [reply CONTRIBUTING.md:135]: I prefer to have something that is clearly defined than a reference to another convention that can change at any time without realizing it.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+```
+<type>[optional scope]: <description>
+
+[optional body]
+
+[optional footer(s)]
+```
+
+Any line of the commit message cannot be longer 100 characters! This allows the message to be easier to read on GitHub as well as in various git tools.
```
Comment (alexis-langlet) [inline CONTRIBUTING.md:55]:
```suggestion
Any line of the commit message cannot be longer than 100 characters! This allows the message to be easier to read on GitHub as well as in various git tools.
```

File: CONTRIBUTING.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (alexis-langlet) [reply CONTRIBUTING.md:19]: I quite liked the "Exceptions to this rule include fixing non-functional source such as code comments, documentation or other supporting files." part

File: CONTRIBUTING.md (hunk 5 above)
Comment (EstebanBAR0N) [reply CONTRIBUTING.md:31]: For us, it does too.

**Comment by GridexX** [comment]
We've updated the file according to the reviews :smile: 
Please tell us if it's good !

**Review by nponsard — APPROVED** [review]

**Review by iverly — APPROVED** [review]

**Pull request merged**